### DISCUSSION TITLE
Know if Manifold::Boolean(xxx, OpType::Subtract) operation has actually done something (changed source manifold)

### DISCUSSION THREAD

**Discussion by titovmaxim** [elalish/manifold#891] [Q&A]
I know Issues is not a place to ask questions. However I failed to find any kind of hit for the question.

I have a manifold from which I consequently subtract other manifolds. Would be nice to know if current operation has change anything (for example it can do nothing if manifolds do not intersect at all) at every step. What is the best way to achieve it?

Currently I compare all vertices (taking into account that their indexes can be changed) and faces (taking into account that they can be "rotated" internally). Not too nice operation for large meshes.

**Comment by titovmaxim** [comment]
P.S. I was rally impressed by manifold library highest quality, clear design and performance. Especially after CGAL & Co. Nothing to compare...

**Comment by pca006132** [comment]
I think mesh ID may work? Or use the `Split` method which computes both the intersection and difference. If intersection is small, `Split` may be quite efficient because creating the intersection mesh will take only a small amount of time in that case. The catch of this is that you will not enjoy the optimization done by our underlying CSG tree data structure that attempts to reorder operations for better performance.

And GitHub discussion is the appropriate place for asking questions :)

**Reply by titovmaxim** [reply]
I've got the same idea about ID.  Using AsOriginal before Boolean to set it. However it is reset to -1 even if nothing is really changed (so no subtraction occurred). No subtraction is done, but indices can be reordered and ID is set to -1.

**Reply by titovmaxim** [reply]
A kind of flag from Boolean() would be nice in this case. I can try ti dig a source code myself in order to find out how to insert it. But was unsure if there are not standard means for that.

**Reply by pca006132** [reply]
One issue is that we are doing lazy evaluation, so not sure how to add this flag.

**Reply by elalish** [reply]
Interesting idea - we may be able to add some simple logic to preserve the `OriginalID` if nothing changed. But I agree that `Split` is a robust way to do this - it only solves the intersections once, so it's pretty efficient. Then you can just check if the intersection `IsEmpty()`.

**Reply by titovmaxim** [reply]
That you for help guys.
